.github/workflows/go-getter.yml

```diff
@@ -1,22 +1,25 @@
 name: go-getter
 
-on: [push]
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
 
 env:
   TEST_RESULTS_PATH: /tmp/test-results
 
 jobs:
 
-  linux-tests:
+  # Basic validation that runs on both PRs and pushes (safe operations)
+  basic-validation:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         go-version:
-          - 1.18
-          - 1.19
-    permissions:
```

Comment:
Also, can we add back the permissions here because the default git token has too much permissions?

Reply:
The `basic-validation` and `linter` jobs no longer use the token and so do not need `permissions`. They are more secure because they separate out the token stuff from non-token stuff. `permissions` are still in effect for `linux-integration-tests` and `windows-integration-tests`, which use it.